apps/src/gamelab/GameLabVisualizationColumn.jsx

```diff
@@ -76,12 +98,12 @@ var GameLabVisualizationColumn = React.createClass({
             height={GAME_HEIGHT}
             onMouseMove={this.onMouseMove}
           >
+            <GridOverlay show={this.props.showGrid} />
```

Comment:
The short answer is that when you're inside a ProtectedDiv, it's not going to update on redux store changes. Your options are (a) ensure that it has the right state on initial render and never needs to update (presumably not applicable if you want to toggle on/off dynamically) (b) move this outside of the ProtectedDiv (possibly difficult) (c) Make the div not protected (while cleaning up the things that necessitated it being protected) or (d) toggle the state using something like jquery code instead of react (generally not recommended).

(c) is probably the most desirable, but not sure how plausible it is.